src/main/java/org/opentripplanner/routing/algorithm/mapping/RaptorPathToItineraryMapper.java

```diff
@@ -291,7 +291,9 @@ private void mapNonTransitLeg(List<Leg> legs, PathLeg<TripSchedule> pathLeg, Tra
                 // TODO OTP2 We use the duration initially calculated for use during routing
                 //      because they do not always match up and we risk getting negative wait times
                 //      (#2955)
-                subItinerary.nonTransitTimeSeconds = pathLeg.duration();
+                assert(subItinerary.legs.size() == 1);
```

Comment:
We do not use `assert`, it need to be enabled by a command line switch, which rarely is used. You can do a `if(...) { throw new IllegalArgumentException("<message>"); }` instead.

I personally like programing by contract, so feel free to promote this to become part of the coding guidelines, but until then we should not use it.